recbole/trainer/trainer.py

```diff
@@ -786,7 +784,7 @@ def pretrain(self, train_data, verbose=True, show_progress=False):
                 self.logger.info(train_loss_output)
             self._add_train_loss_to_tensorboard(epoch_idx, train_loss)
 
-            if (epoch_idx + 1) % self.save_step == 0:
+            if (epoch_idx + 1) % self.save_step == 0 and self.config["local_rank"] == 0:
```

Comment:
Refer to: https://pytorch.org/tutorials/intermediate/ddp_tutorial.html#save-and-load-checkpoints:

> All processes should see same parameters as they all start from same random parameters and gradients are synchronized in backward passes. Therefore, saving it in one process is sufficient.

Reply:
Please double-check if the following code needs to be fixed:

1. In `XGBoostTrainer`
https://github.com/RUCAIBox/RecBole/blob/6d9f74216379ccfac068be9769cb3a2858d4fd14/recbole/trainer/trainer.py#L1130-L1135

2. In `LightGBMTrainer`
https://github.com/RUCAIBox/RecBole/blob/6d9f74216379ccfac068be9769cb3a2858d4fd14/recbole/trainer/trainer.py#L1192-L1197

Thanks!